Detectors/TRD/simulation/src/Trap2CRU.cxx

```diff
@@ -613,32 +611,35 @@ void Trap2CRU::convertTrapData(o2::trd::TriggerRecord const& triggerrecord, cons
         // write 2 tracklet end markers irrespective of there being tracklet data.
         writeTrackletEndMarkers();
         linkwordswritten += 2;
-        // always write 2 digit hc headers, irrespective of whether there are digits or not
-        writeDigitHCHeaders(triggercount, hcid);
-        linkwordswritten += 2;
-        while (mCurrentDigit < enddigitindex && mDigits[mDigitsIndex[mCurrentDigit]].getHCId() == hcid) {
-          // while we are on a single mcm, copy the digits timebins to the array.
-          int digitcounter = 0;
-          int currentROB = mDigits[mDigitsIndex[mCurrentDigit]].getROB();
-          int currentMCM = mDigits[mDigitsIndex[mCurrentDigit]].getMCM();
-          int firstDigitMCM = mCurrentDigit;
-          while (mDigits[mDigitsIndex[mCurrentDigit]].getMCM() == currentMCM &&
-                 mDigits[mDigitsIndex[mCurrentDigit]].getROB() == currentROB &&
-                 mDigits[mDigitsIndex[mCurrentDigit]].getHCId() == hcid) {
-            mCurrentDigit++;
-            digitcounter++;
-            if (mCurrentDigit == enddigitindex) {
-              break;
+
+        if (isCalibTrigger) {
```

Comment:
@bazinski if I send a physics trigger and still write the digit HC headers and raw end markers then the raw reader gets quite confused:
```
[40125:trd-datareader]: [20:41:02][INFO] 2 current endpont : 1 previous end point : 0
[40125:trd-datareader]: [20:41:02][INFO] 1 current endpont : 1 previous end point : 0
[40125:trd-datareader]: [20:41:02][INFO] 2 current endpont : 1 previous end point : 0
[40125:trd-datareader]: [20:41:02][INFO] 2 current endpont : 1 previous end point : 0
[40125:trd-datareader]: [20:41:02][INFO] 2 current stopbit: 5 previous stopbit: 1
[40125:trd-datareader]: [20:41:02][INFO] 1 current stopbit: 5 previous stopbit: 1
[40125:trd-datareader]: [20:41:02][INFO] 2 current stopbit: 5 previous stopbit: 1
[40125:trd-datareader]: [20:41:02][INFO] 2 current stopbit: 5 previous stopbit: 1
```
But the way it is implemented here it works fine

Reply:
It should just reject the data after the tracklet end marker, and then go in search of the next tracklethcheader. you might end up with more warnings/info messages than you would care to deal with.

Reply:
The raw data off the detector does not have digit half chamber headers for physics triggers.